src/backend/import/importTransactions.ts

```diff
@@ -19,27 +19,25 @@ type ScrapingConfig = Config['scraping'];
 const TRANSACTION_STATUS_COMPLETED = 'completed';
 
 export async function scrapeFinancialAccountsAndFetchTransactions(scrapingConfig: ScrapingConfig, startDate: Date, eventPublisher: EventPublisher) {
-  const companyIdToTransactions: Record<string, EnrichedTransaction[]> = {};
-
   const dowloadedChrome = await getChrome(userDataPath, (percent) => emitChromeDownload(eventPublisher, percent));
 
-  const accountsToScrape = scrapingConfig.accountsToScrape.filter((accountToScrape) => accountToScrape.active !== false);
-  const scrapingPromises = accountsToScrape.map(async (accountToScrape) => {
-    const companyId = accountToScrape.key;
-    try {
-      await eventPublisher.emit(EventNames.IMPORTER_START, buildImporterEvent(accountToScrape, { message: 'Importer start' }));
-      const scrapeResult = await fetchTransactions(accountToScrape, startDate, scrapingConfig, eventPublisher, dowloadedChrome);
-      const transactions = await postProcessTransactions(accountToScrape, scrapeResult);
-      companyIdToTransactions[companyId] = transactions;
-      await eventPublisher.emit(EventNames.IMPORTER_END, buildImporterEvent(accountToScrape, { message: 'Importer end', status: AccountStatus.DONE }));
-    } catch (error) {
-      await eventPublisher.emit(EventNames.IMPORTER_ERROR, buildImporterEvent(accountToScrape, {
-        message: 'Importer error', error, status: AccountStatus.ERROR
-      }));
-      throw error;
-    }
-  });
-  await Promise.all(scrapingPromises);
+  const scrapePromises = scrapingConfig.accountsToScrape
+    .filter((accountToScrape) => accountToScrape.active !== false)
+    .map(async (accountToScrape) => ({
+      id: accountToScrape.id,
+      transactions: await fetchTransactions(accountToScrape, startDate, scrapingConfig.showBrowser, eventPublisher, dowloadedChrome)
+    }));
+
+  const promiseResults = await Promise.allSettled(scrapePromises);
+  const companyIdToTransactions = promiseResults
+    .reduce((idToTrxAcc, scrapeRes) => {
+      if (scrapeRes.status === 'fulfilled') {
```

Comment:
If you doing this `if` in a `filter` method, TS will not identify the `scrapeRes` as a type of `FullfilledResult`, and you don't have access to `.value`.